### PR TITLE
Feature/home page

### PR DESCRIPTION
**Pull request by LisaQ04**
Added cart state and action, also added spree prefix.

### CODE REVIEW

File: src/Components/Shop/ProductActions.js [hunk 1]
```diff
@@ -72,4 +72,46 @@ export const setProductFilter = (filter) => dispatch => dispatch(setProductFilte
...
+        }
+    }
+});
+
+export const fetchSpecificProductSuccessAC = (product) => ({
+    type: Types.SHOPPING_GET_SPECIFIC_PRODUCT_SUCCEEDED,
+    productState: {
+        spreeSpecificProductState: {
+            requesting: false,
+            requestError: null,
```
Comment (GeromeSchutte) [inline src/Components/Shop/ProductActions.js]: dont need to reset the requesterror on success, you already do it on init

File: src/Components/ShoppingCart/ShoppingCartActions.js [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+        requestError: null,
+        items: []
+    } 
+});
+
+export const fetchCartSuccessAC = (items) => ({
+    type: Types.FETCH_CART_SUCCESS,
+    cartState: {
+        requesting: false,
+        displayTotal: items.data.attributes.display_total,
```
Comment (GeromeSchutte) [inline src/Components/ShoppingCart/ShoppingCartActions.js]: if you are resetting the cart items on fetch begin, you need to also reset the item count and displaytotal

File: src/Components/Shop/ProductActions.js [hunk 3]
```diff
@@ -72,4 +72,46 @@ export const setProductFilter = (filter) => dispatch => dispatch(setProductFilte
...
+    type: Types.SHOPPING_GET_SPECIFIC_PRODUCT_FAILED,
+    productState: {
+        spreeSpecificProductState: {
+            requesting: false,
+            requestError: error 
+        } 
+    }
+});
+
+export const spree_getProduct = (id) => async (dispatch) => {
```
Comment (GeromeSchutte) [inline src/Components/Shop/ProductActions.js]: on this level you shouldn't add the spree_ prefix :) only to api's. Components do not need to know that they are dealing with spree.

File: src/Components/Shop/ProductActions.js (hunk 1 above)
Comment (LisaQ04) [reply src/Components/Shop/ProductActions.js]: removed request error.

File: src/Components/ShoppingCart/ShoppingCartActions.js (hunk 2 above)
Comment (LisaQ04) [reply src/Components/ShoppingCart/ShoppingCartActions.js]: Now Resetting only request error and requesting on begin.

File: src/Components/Shop/ProductActions.js (hunk 3 above)
Comment (LisaQ04) [reply src/Components/Shop/ProductActions.js]: Changed function name

**Pull request closed without merging**